### PR TITLE
User service unimplemented features

### PR DESCRIPTION
**Pull request by CoderNamedHendrick**
I implemented followerIds and friendIds functions which were marked as @notImplemented.
@jonjomckay @nikolajjsj  @robertodoering  @dre8597

### CODE REVIEW

**Comment by CoderNamedHendrick** [comment]
I've implemented the rest of the user service endpoints asides from two endpoints with a bit of a special case.
So this [endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friendships-no_retweets-ids) returns a list of ids. While I can parse it by writing the fromJson and toJson methods myself, I was wondering if there's a way to do this with the json_serializer.
@robertodoering @nikolajjsj

**Comment by robertodoering** [comment]
I will have a look this weekend, thanks. Btw there is no need to tag everyone, they probably don't want to get notifications for this pr.

**Comment by CoderNamedHendrick** [comment]
Hey @robertodoering, just reminding you about the review.☺️😄

File: lib/api/users/data/paginated_ids.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'paginated_ids.g.dart';
+
+@JsonSerializable(
+  explicitToJson: true,
+  fieldRename: FieldRename.snake,
+  includeIfNull: false,
```
Comment (robertodoering) [inline lib/api/users/data/paginated_ids.dart]: This is not necessary and should be removed for consistency (it only affects the `toJson` output anyways).

If the json response is missing nullable fields, they are always parsed as `null`, even without this flag.

File: lib/api/users/user_service.dart [hunk 2]
```diff
@@ -29,11 +31,25 @@ class UserService {
   /// [usersLookup], a method that allows you to convert user IDs into full user
   /// objects in bulk.
   ///
```
Comment (robertodoering) [inline lib/api/users/user_service.dart:32]: Would be nice to have descriptions for the parameters in the docs, just like the other methods do.

File: lib/api/users/user_service.dart [hunk 3]
```diff
@@ -175,11 +205,19 @@ class UserService {
...
-  Future<void> friendshipsIncoming() async {}
+  Future<PaginatedIds> friendshipsIncoming({
+    int? cursor,
+    TransformResponse<PaginatedIds> transform = defaultPaginatedIdsTransform,
+  }) async {
+    final params = <String, String>{}..addParameter('cursor', cursor);
+
+    return client
+        .get(Uri.https(
+            'api.twitter.com', '1.1/friendships/incoming.json', params))
```
Comment (robertodoering) [inline lib/api/users/user_service.dart]:
```suggestion
            'api.twitter.com', '1.1/friendships/incoming.json', params),)
```

File: lib/api/users/data/relationship_entity.dart [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'relationship_entity.g.dart';
+
+@JsonSerializable(
+  explicitToJson: true,
+  fieldRename: FieldRename.snake,
+  includeIfNull: false,
+)
+class RelationshipEntity {
```
Comment (robertodoering) [inline lib/api/users/data/relationship_entity.dart:9]: This model is missing a `bool? muting` field
See example response in https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friendships-show

**Comment by CoderNamedHendrick** [comment]
> I've implemented the rest of the user service endpoints asides from two endpoints with a bit of a special case. So this [endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friendships-no_retweets-ids) returns a list of ids. While I can parse it by writing the fromJson and toJson methods myself, I was wondering if there's a way to do this with the json_serializer. @robertodoering

**Comment by robertodoering** [comment]
> I've implemented the rest of the user service endpoints asides from two endpoints with a bit of a special case. So this [endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friendships-no_retweets-ids) returns a list of ids. While I can parse it by writing the fromJson and toJson methods myself, I was wondering if there's a way to do this with the json_serializer. @robertodoering

You can write the fromJson yourself.
I would add a `Future<List<int>> defaultIntListTransform(String body)` method along with the isolate wrapper in `transforms.dart` and then decode the body manually.

It'll look something like this
```dart
return (body as List<dynamic>)
        .map((e) => e as int)
        .toList();
```

**Comment by CoderNamedHendrick** [comment]
That makes sense, thanks.

**Review by robertodoering — APPROVED** [review]
Looks great, thanks for your contribution :)

**Comment by CoderNamedHendrick** [comment]
Awesome, thank you so much for this opportunity.😃

**Comment by CoderNamedHendrick** [comment]
@robertodoering it seems the workflow needs your approval to run.

**Comment by CoderNamedHendrick** [comment]
Oh sorry, my bad😅

I was trying to get yours or any other contributor’s attention, also I left
a comment concerning why I haven’t finished with the user services. Thanks
again for your help Robert, I’m grateful.

**Pull request merged**